test/azure/availabilityset_test.go

```diff
@@ -0,0 +1,74 @@
+// +build azure
```

Comment:
Typically, the tests we put in the `test` package are tests for the example code in the `examples` folder (e.g., Terraform examples). We see these as "integration" tests.

If you have "unit" test that calls some method directly, without any other example code in between (e.g., no Terraform examples), then that should go into the same package as the method you're testing: e.g., if you're unit testing a method in `modules/azure/foo.go`, the tests go into `modules/azure/foo_test.go`.

Reply:
Note that this was a decision we made to move all azure tests to the `test` package.

See https://github.com/gruntwork-io/terratest/pull/594#discussion_r483192207 for the relevant discussion.

I was on the fence, but decided that it was probably worth experimenting this style in Azure modules given that I personally always found it confusing to have these "unit" tests since it is almost always better to launch infrastructure with terraform than go.

Reply:
Hm, I think we want to keep the `test` folder solely for "integration tests" that have copy/pasteable examples for users. We don't want all our unit tests mixed up in here, as those are not something users would take advantage of...

Reply:
This was an area we were not sure what the right approach was.  In summary, we should move unit test back under modules folder. as @yorinasub17 mentioned.  Will work with team on that.

Reply:
Fair enough and I am onboard with porting that back.

Side note:

> as those are not something users would take advantage of...

I think I keep coming back to the desire to have `examples-for-humans` and `examples-for-machines`, and `tests-for-humans` and `tests-for-machines`. Right now we are overloading the `examples` and `tests` folder to serve both purposes: for learning, and for testing the repo. I think we should call out which examples and tests are specifically for learning, and map out a learning path with increasingly complex examples.

I just find it awkward to create infrastructure in golang like we are doing in the `aws` module to test a bunch of things in unit, because we want to avoid polluting the `examples` folder which we view as copy pastable by humans. As mentioned above, it feels much cleaner to rely on `terraform` to create infrastructure than awkward SDK calls in golang...

Reply:
Unit tests reverted to modules/azure